sklbench/runner/implementation.py

```diff
@@ -0,0 +1,133 @@
+    return_code = 0
+    bench_cases_with_pbar = tqdm(bench_cases)
+    for bench_case in bench_cases_with_pbar:
+        bench_cases_with_pbar.set_description(
+            custom_format(
+                get_bench_case_name(bench_case, shortened=True), bcolor="HEADER"
+            )
+        )
+        try:
+            bench_return_code, bench_entries = run_benchmark_from_case(
+                bench_case, filters, log_level
+            )
+            if bench_return_code != 0:
+                return_code = bench_return_code
+                if early_exit:
+                    break
+            for entry in bench_entries:
+                entry["environment_name"] = environment_name
+                results.append(entry)
+        except KeyboardInterrupt:
+            return_code = -1
+            break
+    full_result = {
+        "bench_cases": results,
+        "environment": {environment_name: env_info},
+    }
+    return return_code, full_result
+
+
+def run_benchmarks(args: argparse.Namespace) -> int:
+    # overwrite all logging levels if requested
+    if args.log_level is not None:
+        for log_type in ["runner", "bench", "report"]:
+            setattr(args, f"{log_type}_log_level", args.log_level)
+    # set logging level
+    logger.setLevel(args.runner_log_level)
+
+    # find and parse configs
+    bench_cases = generate_bench_cases(args)
+
+    # get parameter filters
+    param_filters = generate_bench_filters(args.parameter_filters)
+
+    # perform early filtering based on 'data' parameters and
+    # some of 'algorithm' parameters assuming they were already assigned
+    bench_cases = early_filtering(bench_cases, param_filters)
+
+    # prefetch datasets
+    if args.prefetch_datasets:
+        # trick: get unique dataset names only to avoid loading of same dataset
+        # by different cases/processes
+        dataset_cases = {get_data_name(case): case for case in bench_cases}
+        logger.debug(f"Unique dataset names to load:\n{list(dataset_cases.keys())}")
+        n_proc = min([16, cpu_count(), len(dataset_cases)])
+        logger.info(f"Prefetching datasets with {n_proc} processes")
+        with Pool(n_proc) as pool:
+            pool.map(load_data, dataset_cases.values())
+
+    # run bench_cases
+    return_code, result = call_benchmarks(
+        bench_cases,
+        param_filters,
+        args.bench_log_level,
+        args.environment_name,
+        args.exit_on_error,
+    )
+
+    # output as pandas dataframe
+    if len(result["bench_cases"]) != 0:
+        for key, df in get_result_tables_as_df(result).items():
+            logger.info(f'{custom_format(key, bcolor="HEADER")}\n{df}')
+
+    # output raw result
+    logger.debug(custom_format(result))
+
+    with open(args.result_file, "w") as fp:
+        json.dump(result, fp, indent=4)
+
+    # generate report
+    if args.report:
+        if args.result_file not in args.result_files:
+            args.result_files += [args.result_file]
```

Comment:
When running benchmarks with the `--report` flag (like [this](https://github.com/IntelPython/scikit-learn_bench/pull/133/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R62)), we set our own value for the `args.result_file` argument and not providing a value for `args.result_files`, whose default value `["result.json"]` is still used in this block of code. As a result, we get an error saying that `result.json` is not found when the value of `args.result_file` is different from `result.json`